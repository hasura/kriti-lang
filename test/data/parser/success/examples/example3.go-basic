[
  {{ $.path }},
  {{ $foo.path }},
  {{ x.path.foo }},
  {{ x.path.foo[1] }}
]
