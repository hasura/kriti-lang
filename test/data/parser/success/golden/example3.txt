Array
    [ Path
        [
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 2
                    , _column = 6
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 2
                        , _column = 7
                        }
                    )
                )
            , Obj "$"
            )
        ,
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 2
                    , _column = 7
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 2
                        , _column = 11
                        }
                    )
                )
            , Obj "path"
            )
        ]
    , Path
        [
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 3
                    , _column = 6
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 3
                        , _column = 10
                        }
                    )
                )
            , Obj "$foo"
            )
        ,
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 3
                    , _column = 10
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 3
                        , _column = 14
                        }
                    )
                )
            , Obj "path"
            )
        ]
    , Path
        [
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 4
                    , _column = 6
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 4
                        , _column = 7
                        }
                    )
                )
            , Obj "x"
            )
        ,
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 4
                    , _column = 7
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 4
                        , _column = 11
                        }
                    )
                )
            , Obj "path"
            )
        ,
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 4
                    , _column = 12
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 4
                        , _column = 15
                        }
                    )
                )
            , Obj "foo"
            )
        ]
    , Path
        [
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 5
                    , _column = 6
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 5
                        , _column = 7
                        }
                    )
                )
            , Obj "x"
            )
        ,
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 5
                    , _column = 7
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 5
                        , _column = 11
                        }
                    )
                )
            , Obj "path"
            )
        ,
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 5
                    , _column = 12
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 5
                        , _column = 15
                        }
                    )
                )
            , Obj "foo"
            )
        ,
            (
                ( SourcePosition
                    { _sourceName = "sourceName"
                    , _line = 5
                    , _column = 16
                    }
                , Just
                    ( SourcePosition
                        { _sourceName = "sourceName"
                        , _line = 5
                        , _column = 18
                        }
                    )
                )
            , Arr 1
            )
        ]
    ]